### PR TITLE
Fix Float input fields Java codegen with default values in json schemas

### PR DESCRIPTION
**Pull request by joaquim-verges**
see https://github.com/apollographql/apollo-android/issues/3108

### CODE REVIEW

File: apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/ir/IRBuilder.kt [hunk 1]
```diff
@@ -139,35 +139,34 @@ class IRBuilder(private val schema: IntrospectionSchema,
...
+      is GraphQLParser.NumberValueContext ->
+        when (ScalarType.forName(type.name ?: "")) {
+          ScalarType.INT -> NUMBER().text.toInt()
+          ScalarType.FLOAT -> NUMBER().text.toDouble()
+          else -> throw ParseException("Unexpected number type")
+        }
       is GraphQLParser.ArrayValueContext -> {
         this.arrayValueType().valueOrVariable()?.map {
-          it.value()?.toKotlin()
+          it.value()?.toKotlin(type)
```
Comment (martinbonnin) [inline apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/ir/IRBuilder.kt:169]: For lists, this should "unwrap" the item type with `type.ofType` as it recurses in order to coerce scalar items appropriately. For an example with `defaultValue = "[3]"`, `type.kind == LIST` and `type.ofType.kind == SCALAR`. Same thing for input fields where it gets even more complicated as we'll need a reference to the schema to lookup the inputFieldDefinition.

**Comment by martinbonnin** [comment]
Thanks for catching this! I believe there are edge cases around lists/input objects where nested values wouldn't get the proper type. Let me know if you want to keep digging ⛏️ . Or I can also try to fix this as I introduced that bug in the first place.

**Comment by joaquim-verges** [comment]
@martinbonnin yep this doesn't cover the nested values case, but does fix the plain float scalar case at least 😅 

Feel free to add another change on top of this PR (I think you can?), or do a follow up. 

Either way I'll let you decide what you want to do with this PR.

Thanks!

**Comment by martinbonnin** [comment]
> Feel free to add another change on top of this PR (I think you can?)

Will do, thank you so much for the contribution !

**Pull request merged**